src/sql/workbench/contrib/notebook/browser/models/diffNotebookInput.ts

```diff
@@ -0,0 +1,91 @@
+import { ILogService } from 'vs/platform/log/common/log';
+
+export class DiffNotebookInput extends SideBySideEditorInput {
+	public static ID: string = 'workbench.editorinputs.DiffNotebookInput';
+	private _notebookService: INotebookService;
+	private _logService: ILogService;
+
+	constructor(
+		title: string,
+		diffInput: DiffEditorInput,
+		@IInstantiationService instantiationService: IInstantiationService,
+		@INotebookService notebookService: INotebookService,
+		@ILogService logService: ILogService
+	) {
+		let originalInput = instantiationService.createInstance(FileNotebookInput, diffInput.primary.getName(), diffInput.primary.resource, diffInput.originalInput as FileEditorInput);
+		let modifiedInput = instantiationService.createInstance(FileNotebookInput, diffInput.secondary.getName(), diffInput.secondary.resource, diffInput.modifiedInput as FileEditorInput);
+		super(title, diffInput.getTitle(), modifiedInput, originalInput);
+		this._notebookService = notebookService;
+		this._logService = logService;
+		this.setupScrollListeners(originalInput, modifiedInput);
+	}
+
+	public getTypeId(): string {
+		return DiffNotebookInput.ID;
+	}
+
+	/**
+	 * Setup scroll listeners so that both the original and modified editors scroll together
+	 * @param originalInput original notebook input
+	 * @param modifiedInput modified notebook input
+	 */
+	private setupScrollListeners(originalInput: FileNotebookInput, modifiedInput: FileNotebookInput): void {
+		Promise.all([originalInput.containerResolved, modifiedInput.containerResolved]).then(() => {
+
+			// Setting container height to 100% ensures that scrollbars will be added when in diff mode
+			originalInput.container.parentElement.style.height = '100%';
+			modifiedInput.container.parentElement.style.height = '100%';
+
+			// Keep track of when original and modified notebooks are shown
+			const originalNotebookEditorShown: Deferred<void> = new Deferred<void>();
+			const modifiedNotebookEditorShown: Deferred<void> = new Deferred<void>();
+
+			// Possible for notebooks to have been shown already, so check this case
+			if (this._notebookService.findNotebookEditor(originalInput.notebookUri)) {
+				originalNotebookEditorShown.resolve();
+			} else if (this._notebookService.findNotebookEditor(modifiedInput.notebookUri)) {
+				modifiedNotebookEditorShown.resolve();
+			}
+
+			// If not already shown, listen for add events
+			this._register(this._notebookService.onNotebookEditorAdd((e) => {
+				if (e.id === originalInput.notebookUri.toString()) {
+					originalNotebookEditorShown.resolve();
+				} else if (e.id === modifiedInput.notebookUri.toString()) {
+					modifiedNotebookEditorShown.resolve();
+				}
+			}));
+
+			// Once both are shown, look for scrollable DIV. Add scroll listeners here
+			Promise.all([originalNotebookEditorShown.promise, modifiedNotebookEditorShown.promise]).then(() => {
+				const originalScrollableNode = originalInput.container?.querySelector('.scrollable');
```

Comment:
I tested out the scrolling and it's really slow for me - not the normal scrolling speed. I assume this isn't happening for you? Let me know if you want help debugging that since it's a pretty bad experience right now when that happens

Reply:
I don't have slow scrolling at all. That's bizarre. Hmm.